### PR TITLE
Feature: Add support for pausing/resuming subscriptions

### PR DESCRIPTION
**Pull request by pauloiankoski**
Resolves [GIVE-1250]

## Description
This Pull Request adds supports for gateways to implement their solutions for pausing/resuming subscriptions. Within this PR we are:
- adding a new "paused" status to subscriptions;
- setting a color for the "paused" status in the subscriptions list table;
- adding a new `SubscriptionPausable` contract to be signed by gateways;
- adding a capability check as `can_pause` to gateways (false by default if not implemented);
- fixing the `SubscriptionRepository` to update the expiration date into the database;
- updating the subscription details screen to show the next renewal date for paused subscriptions as it does for active ones.

This PR is a foundational work that is required for the new implementations on Give Recurring.

## Affects
Subscriptions

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-1250]: https://stellarwp.atlassian.net/browse/GIVE-1250?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Framework/PaymentGateways/SubscriptionModule.php [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 use Give\Framework\PaymentGateways\Contracts\Subscription\SubscriptionAmountEditable;
 use Give\Framework\PaymentGateways\Contracts\Subscription\SubscriptionDashboardLinkable;
+use Give\Framework\PaymentGateways\Contracts\Subscription\SubscriptionPausable;
 use Give\Framework\PaymentGateways\Contracts\Subscription\SubscriptionPaymentMethodEditable;
+use Give\Framework\PaymentGateways\Contracts\Subscription\SubscriptionResumable;
```
Comment (JoshuaHungDinh) [inline src/Framework/PaymentGateways/SubscriptionModule.php:9]: Do we need this still?

Comment (pauloiankoski) [reply src/Framework/PaymentGateways/SubscriptionModule.php:9]: No. That's a leftover.

Comment (JoshuaHungDinh) [reply src/Framework/PaymentGateways/SubscriptionModule.php:9]: Just checking, everything seems to be in order with the UI and this work. This should be good to go! Would you prefer I make an epic for the UI & this or would you rather work off dev?

**Review by JoshuaHungDinh — APPROVED** [review]
Nice!

**Comment by jonwaldstein** [comment]
@pauloiankoski we decided to loosen the restriction on the `SubscriptionPausable` contract to receive an array of data instead of `intervalInMonths` - reason being want this to be flexible for other gateways and since this is our first gateway it's hard to predict what they will need.

**Comment by pauloiankoski** [comment]
Cool. That's something I was in doubt. I see other methods receiving a generic `$gatewayData` array instead.

**Comment by jonwaldstein** [comment]
@pauloiankoski yeah it's just hard to predict what params would be helpful for _all gateways_ at this point, so i'd rather keep it flexible as changing signatures on contracts is painful 😄

**Pull request merged**